nomenclature/region_mapping_models.py

```diff
@@ -150,3 +150,34 @@ def create_from_region_mapping(cls, file: Union[Path, str]):
                 )
             mapping_input["common_regions"] = common_region_list
         return cls(**mapping_input)
+
+
+class ModelMappingCollisionError(ValueError):
+    template = "Model {model} is defined twice in {files}"
+
+    def __init__(
+        self, mapping1: RegionAggregationMapping, mapping2: RegionAggregationMapping
+    ) -> None:
+        files = (
+            mapping1.file.relative_to(Path.cwd()),
+            mapping2.file.relative_to(Path.cwd()),
+        )
+        super().__init__(self.template.format(model=mapping1.model, files=files))
+
+
+class RegionProcessor(BaseModel):
+    """Reads region mappings from a directory and uses them for region aggregation"""
+
+    mappings: Dict[str, RegionAggregationMapping]
+
+    @classmethod
+    @validate_arguments
+    def from_directory(cls, mappingdir: DirectoryPath):
```

Comment:
I'd call this argument `path` following the API of the `CodeList`, see
https://github.com/IAMconsortium/nomenclature/blob/057c8922deda3fa79639e9048f9c9c67ccb0d6e4/nomenclature/codes.py#L56

Reply:
Done